### PR TITLE
[freeimage] Enable optional plugins; fixes #9894

### LINKED ISSUE #9884: [package]  libx264/20191217: can't build on ubuntu
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libx264/20191217**
  * Operating System+version: **Linux Ubuntu 20.04**
  * Compiler+version: **GCC 11.2**
  * Conan version: **conan 1.46.2**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=11.2
os=Linux
os_build=Linux
[options]
*:shared=False
vulkan-loader:shared=True
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
conan install libx264/20191217@ --build missing -u -r conancenter

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
$ conan install libx264/20191217@ --build missing -u -r conancenter
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=gcc
compiler.cppstd=17
compiler.libcxx=libstdc++11
compiler.version=11.2
os=Linux
os_build=Linux
[options]
*:shared=False
vulkan-loader:shared=True
[build_requires]
[env]

libx264/20191217: Retrieving from server 'conancenter' 
libx264/20191217: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.10k]                                          
Downloading conanfile.py completed [6.80k]                                               
Downloading conan_export.tgz completed [0.26k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
libx264/20191217: Downloaded recipe revision 7b1b8a2a400fa56a8884273bc3145684
nasm/2.15.05: WARN: Current package is newer than remote upstream one
Installing package: libx264/20191217
Requirements
    libx264/20191217 from 'conancenter' - Downloaded
Packages
    libx264/20191217:893dc65bb9cfdc8cca3bc09908d0644d7e00cbb8 - Build
Build requirements
    nasm/2.15.05 from 'conancenter' - Cache
Build requirements packages
    nasm/2.15.05:cc9f9b2b9f92fc82888026414560bf79d8c3a052 - Cache

Installing (downloading, building) binaries...
nasm/2.15.05: Already installed!
nasm/2.15.05: Appending PATH environment variable: /home/andrei/.conan/data/nasm/2.15.05/_/_/package/cc9f9b2b9f92fc82888026414560bf79d8c3a052/bin
libx264/20191217: Applying build-requirement: nasm/2.15.05
libx264/20191217: Configuring sources in /home/andrei/.conan/data/libx264/20191217/_/_/source
Downloading x264-snapshot-20191217-2245.tar.bz2 completed [755.94k]                                

libx264/20191217: Copying sources to build folder
libx264/20191217: Building your package in /home/andrei/.conan/data/libx264/20191217/_/_/build/893dc65bb9cfdc8cca3bc09908d0644d7e00cbb8
libx264/20191217: Generator txt created conanbuildinfo.txt
libx264/20191217: Aggregating env generators
libx264/20191217: Calling build()
libx264/20191217: WARN: Error running `configure --help`: Error 1 while executing source_subfolder/configure --help
libx264/20191217: Calling:
 > source_subfolder/configure '--bit-depth=all' '--disable-cli' '--prefix=/home/andrei/.conan/data/libx264/20191217/_/_/package/893dc65bb9cfdc8cca3bc09908d0644d7e00cbb8' '--enable-static' '--enable-pic' '--enable-debug' 
Found NASM version 2.15.05 compiled on Mar 22 2022
Minimum version is nasm-2.13
If you really want to compile without asm, configure with --disable-asm.
libx264/20191217: 
libx264/20191217: ERROR: Package '893dc65bb9cfdc8cca3bc09908d0644d7e00cbb8' build failed
libx264/20191217: WARN: Build folder /home/andrei/.conan/data/libx264/20191217/_/_/build/893dc65bb9cfdc8cca3bc09908d0644d7e00cbb8
ERROR: libx264/20191217: Error in build() method, line 149
	autotools = self._configure_autotools()
while calling '_configure_autotools', line 140
	self._autotools.configure(args=args, vars=self._override_env, configure_dir=self._source_subfolder, build=build_canonical_name, host=host_canonical_name)
	ConanException: Error 1 while executing source_subfolder/configure '--bit-depth=all' '--disable-cli' '--prefix=/home/andrei/.conan/data/libx264/20191217/_/_/package/893dc65bb9cfdc8cca3bc09908d0644d7e00cbb8' '--enable-static' '--enable-pic' '--enable-debug' 

```

</details>

### LINKED ISSUE #9894: [package] freeimage/3.18.0: Plugins aren't enabled
The preprocessor symbols used in the patches (WITH_JPEG etc.) are not defined by cmake.

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **freeimage/3.18.0**
  * Operating System+version: **Windows 11**
  * Compiler+version: **Visual Studio 2019**
  * Conan version: **conan 1.46.2**
  * Python version: **Python 3.10.3**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
```

### Steps to reproduce (Include if Applicable)

Load image with any optional plugin (JPEG, PNG, TIFF, etc).  The test package didn't check for errors.

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
freeimage/3.18.0 (test package): Running test()
FreeImage 3.18.0, with:
	- bmp
	- ico
	- jng
	- koa
	- iff,lbm
	- mng
	- pbm
	- pbm
	- pcd
	- pcx
	- pgm
	- pgm
	- ppm
	- ppm
	- ras
	- tga,targa
	- wap,wbmp,wbm
	- psd,psb
	- cut
	- xbm
	- xpm
	- dds
	- gif
	- hdr
	- sgi,rgb,rgba,bw
	- pfm
	- pct,pict,pic
```

Should be:

```
freeimage/3.18.0 (test package): Running test()
FreeImage 3.18.0, with:
	- bmp
	- ico
	- jpg,jif,jpeg,jpe
	- jng
	- koa
	- iff,lbm
	- mng
	- pbm
	- pbm
	- pcd
	- pcx
	- pgm
	- pgm
	- png
	- ppm
	- ppm
	- ras
	- tga,targa
	- tif,tiff
	- wap,wbmp,wbm
	- psd,psb
	- cut
	- xbm
	- xpm
	- dds
	- gif
	- hdr
	- sgi,rgb,rgba,bw
	- exr
	- j2k,j2c
	- jp2
	- pfm
	- pct,pict,pic
	- 3fr,arw,bay,bmq,cap,cine,cr2,crw,cs1,dc2,dcr,drf,dsc,dng,erf,fff,ia,iiq,k25,kc2,kdc,mdc,mef,mos,mrw,nef,nrw,orf,pef,ptx,pxn,qtk,raf,raw,rdc,rw2,rwl,rwz,sr2,srf,srw,sti,x3f
	- webp
	- jxr,wdp,hdp
```

</details>

### PR DESCRIPTION
**Pull request by gmeeker**
Specify library name and version:  **freeimage/3.18.0**

Preprocessor definitions expected by the freeimage patches were missing in CMake.  Also the test package didn't check for errors.

fixes #9884 
fixes #9895

---

- [X ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/freeimage/all/test_package/test_package.cpp [hunk 1]
```diff
@@ -32,15 +34,25 @@ int main(int argc, char** argv )
...
+        else
+        {
+            std::cerr << "Unable to load " << image_file << std::endl;
+            return 10;
+        }
+    }
+    else
+    {
+        std::cerr << "Missing plugin for " << image_file << std::endl;
+        return 10;
```
Comment (intelligide) [inline recipes/freeimage/all/test_package/test_package.cpp:55]: the test_package is just there to validate that the package is correctly created, and that package consumers will be able to link against it. This is not meant to test if the library is working properly. It just needs to be able to be built and run without link errors.

Comment (gmeeker) [reply recipes/freeimage/all/test_package/test_package.cpp]: Okay, I've reverted that section.

**Review by intelligide — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by SpaceIm** [comment]
> fixes #9884 fixes #9895

How is it related to #9884?

**Comment by SpaceIm** [comment]
> Preprocessor definitions expected by the freeimage patches were missing in CMake. Also the test package didn't check for errors.

I don't understand. Where do you see this preprocessor definitions in freeimage patches?

**Pull request merged**